File: Data Structures/LinearQueue.java
public class LinearQueue {
	private Double[] queueArray;
	private int maxSize;
	private int front; // tracks the front value of the queue
	private int rear; // tracks the rear value of the queue
	int nItems; // Tracks the no of items in the queue

	public LinearQueue(int max) { // constructor
		this.maxSize = max;
		queueArray = new Double[max];
		front = 0;
		rear = -1;
		nItems = 0;
	}

	public boolean isEmpty() {
		return (nItems == 0);
	}

	public boolean isFull() {
		return (rear == maxSize - 1);
	}

	public void enqueue(double value) {
		if (isFull()) {
			System.out.println("Sorry the Queue is Full");
		} else {
			queueArray[++rear] = value;
			nItems++;
		}
	}

	public double dequeue() {
		if (isEmpty()) {
			return -9999;
		} else {
			nItems--;
			return queueArray[front++];
		}
	}

	public double peek() {
		if (isEmpty()) {
			return -9999;
		} else {
			return queueArray[front];
		}
	}

}

File: Data Structures/LinearQueue.cs
public class LinearQueue
{
    private double[] queueArray;
    private int maxSize;
    private int front; // Tracks the front value of the queue
    private int rear;  // Tracks the rear value of the queue
    private int nItems; // Tracks the number of items in the queue

    public LinearQueue(int max) // Constructor
    {
        this.maxSize = max;
        queueArray = new double[maxSize];
        front = 0;
        rear = -1;
        nItems = 0;
    }

    public bool IsEmpty() // Check if the queue is empty
    {
        return (nItems == 0);
    }

    public bool IsFull() // Check if the queue is full
    {
        return (rear == maxSize - 1);
    }

    public void Enqueue(double value) // Add an item to the queue
    {
        if (IsFull())
        {
            Console.WriteLine("Sorry, the Queue is Full");
        }
        else
        {
            queueArray[++rear] = value;
            nItems++;
        }
    }

    public double Dequeue() // Remove and return the front item of the queue
    {
        if (IsEmpty())
        {
            return -9999; // Return an error value if the queue is empty
        }
        else
        {
            nItems--;
            return queueArray[front++];
        }
    }

    public double Peek() // Return the front item without removing it
    {
        if (IsEmpty())
        {
            return -9999; // Return an error value if the queue is empty
        }
        else
        {
            return queueArray[front];
        }
    }
}
